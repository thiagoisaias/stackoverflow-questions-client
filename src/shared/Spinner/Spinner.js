import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Container = styled.div`
  width: ${props => `${props.size}px`};
  height: ${props => `${props.size}px`};
  margin: 0px auto;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const StyledSpinner = styled.div`
  display: inline-block;
  position: relative;
  width: 100%;
  height: 100%;

  > div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 100%;
    height: 100%;
    border: 4px solid;
    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: ${props =>
      `${props.color} transparent transparent transparent`};
  }
  > div:nth-child(1) {
    animation-delay: -0.45s;
  }
  > div:nth-child(2) {
    animation-delay: -0.3s;
  }
  > div:nth-child(3) {
    animation-delay: -0.15s;
  }

  @keyframes lds-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const Spinner = props => {
  const { size, color } = props;
  return (
    <Container size={size}>
      <StyledSpinner color={color}>
        <div />
        <div />
        <div />
        <div />
      </StyledSpinner>
    </Container>
  );
};

Spinner.propTypes = {
  size: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired
};

export default Spinner;
