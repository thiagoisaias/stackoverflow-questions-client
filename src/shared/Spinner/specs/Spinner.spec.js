import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Spinner from "../Spinner";

describe("Spinner", () => {
  const mockProps = {
    size: 30,
    color: "#FFF"
  };
  const component = shallow(<Spinner {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
