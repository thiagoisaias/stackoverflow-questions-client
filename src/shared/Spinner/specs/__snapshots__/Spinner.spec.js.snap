// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders properly 1`] = `
<styled.div
  size={30}
>
  <styled.div
    color="#FFF"
  >
    <div />
    <div />
    <div />
    <div />
  </styled.div>
</styled.div>
`;
